crates/iceberg/src/spec/mod.rs

```diff
@@ -17,7 +17,20 @@
 
 //! Spec for Iceberg.
 
-pub mod datatypes;
-pub mod schema;
-pub mod transform;
-pub mod values;
+mod datatypes;
+mod partition;
+mod schema;
+mod snapshot;
+mod sort;
+mod table_metadata;
+mod transform;
+mod values;
+
+pub use datatypes::{ListType, MapType, NestedField, PrimitiveType, StructType, Type};
+pub use partition::{PartitionField, PartitionSpec};
+pub use schema::{visit_schema, visit_struct, visit_type, Schema, SchemaVisitor};
+pub use snapshot::{Operation, Reference, Snapshot, Summary};
+pub use sort::{SortField, SortOrder};
+pub use table_metadata::TableMetadata;
+pub use transform::Transform;
+pub use values::{Literal, PrimitiveLiteral, Struct};
```

Comment:
How about simply make it "pub use schema::*; pub use partition::*;"

Reply:
I prefer manually controlling the exported type to avoid unexpected exports. Additionally, we can safely use `pub struct SchemaV1` internally.

Reply:
I'm not sure if manually controlling the exported type would be the best practice since it introduces extra maintains effort. If we want to limit types exposed, we could use access modifier like `pub(crate) struct SchemaV1`

Reply:
Both LGTM.

Reply:
Then I will use `pub use schema::*;`